descartes_core/include/descartes_core/path_planner_base.h

```diff
@@ -0,0 +1,53 @@
+/*
+ * path_planner_base.h
+ *
+ *  Created on: Jan 19, 2015
+ *      Author: ros developer 
+ */
+
+#ifndef DESCARTES_CORE_PATH_PLANNER_BASE_H_
+#define DESCARTES_CORE_PATH_PLANNER_BASE_H_
+
+#include <descartes_core/trajectory_pt.h>
+#include <descartes_core/robot_model.h>
+#include <vector>
+
+namespace descartes_core
+{
+
+class PathPlannerBase
+{
+public:
+  virtual ~PathPlannerBase(){}
+
+  /**
+   * @brief Plans a path for the given robot model through the points in the input trajectory.
+   * @param model robot model implementation for which to plan a path
+   * @param sampling This value will be interpreted differently by each planner specialization.
```

Comment:
While it can be interpreted differently, it must have a similar use.  We must be more specific here.